### PR TITLE
Create instance if it doesn't exist

### PR DESCRIPTION
**Pull request by CarlSchwan**

### CODE REVIEW

File: lib/Db/InstancesRequest.php [hunk 1]
```diff
@@ -54,7 +55,19 @@ public function save(Instance $instance) {
...
+			->setValue('title', $qb->createNamedParameter($instance->getTitle()))
+			->setValue('version', $qb->createNamedParameter($instance->getVersion()))
+			->setValue('short_description', $qb->createNamedParameter($instance->getShortDescription()))
+			->setValue('description', $qb->createNamedParameter($instance->getDescription()))
+			->setValue('email', $qb->createNamedParameter($instance->getEmail()))
+			->setValue('urls', $qb->createNamedParameter(json_encode($instance->getUrls())))
+			->setValue('stats', $qb->createNamedParameter(json_encode($instance->getStats())))
+			->setValue('usage', $qb->createNamedParameter(json_encode($instance->getUsage())))
+			->setValue('image', $qb->createNamedParameter($instance->getImage()))
+			->setValue('languages', $qb->createNamedParameter(json_encode($instance->getImage())))
```
Comment (CarlSchwan) [inline lib/Db/InstancesRequest.php:69]:
```suggestion
			->setValue('languages', $qb->createNamedParameter(json_encode($instance->getLanguages())))
```

File: lib/Service/InstanceService.php [hunk 2]
```diff
@@ -36,33 +36,40 @@
...
-	public function createLocal(): void {
+	public function createLocal(): Instance {
+		$instance = new Instance();
+		$instance->setLocal(true)
+			->setVersion($this->config->getAppValue('social', 'installed_version', '0.0'))
+			->setApprovalRequired(false)
+			->setDescription($this->config->getAppValue('theming', 'slogan', 'a safe home for your data'))
+			->setTitle($this->config->getAppValue('theming', 'name', 'Nextcloud Social'));
+		$this->instancesRequest->save($instance);
+		return $instance;
```
Comment (ArtificialOwl) [inline lib/Service/InstanceService.php:69]: we need to implement on the front-end a way to assign a contactAccount to the `Instance`

Comment (CarlSchwan) [reply lib/Service/InstanceService.php:69]: And also to add all the other fields. Could you create an issue so that we don't forget?

**Review by ArtificialOwl — APPROVED** [review]

File: lib/Service/InstanceService.php (hunk 2 above)
Comment (ArtificialOwl) [reply lib/Service/InstanceService.php:71]: done

**Pull request merged**